crates/bcr-wdc-quote-client/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+// ----- standard library imports
+// ----- extra library imports
+use bcr_ebill_core::{bill::BitcreditBill, contact::BillParticipant};
+use bcr_wdc_webapi::bill as web_bill;
+use bcr_wdc_webapi::quotes as web_quotes;
+pub use reqwest::Url;
+use thiserror::Error;
+use uuid::Uuid;
+// ----- local imports
+
+// ----- end imports
+
+pub type Result<T> = std::result::Result<T, Error>;
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("resource not found {0}")]
+    ResourceNotFound(Uuid),
+    #[error("invalid request")]
+    InvalidRequest,
+    #[error("Schnorr signing error")]
+    Signing(#[from] bcr_wdc_utils::keys::SchnorrBorshMsgError),
+
+    #[error("internal error {0}")]
+    Reqwest(#[from] reqwest::Error),
+}
+
+#[derive(Debug, Clone)]
+pub struct QuoteClient {
+    cl: reqwest::Client,
+    base: reqwest::Url,
+}
+
+impl QuoteClient {
+    pub fn new(base: reqwest::Url) -> Self {
+        Self {
+            cl: reqwest::Client::new(),
+            base,
+        }
+    }
+
+    pub async fn enquire(
+        &self,
+        bill: BitcreditBill,
```

Comment:
Hm trying to integrate it, I'm not sure this is such a good idea to rely on the core types, since if the version used here is different than the one in ebill-api, this doesn't work.

Reply:
Also, we use different date formats for maturity date (2025-01-01 on our side, rfc3339 on yours)

Reply:
I think the API with the `BillInfo` before was fine actually :+1: